### PR TITLE
Translate project name

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
Don't set a project name in the fs unless the user does, so we can
default it in the UI layer based on the current language.

See #450

### CODE REVIEW

File: src/e2e/app.ts [hunk 1]
```diff
@@ -272,12 +272,12 @@ export class App {
...
   }
 
-  async openSettingsDialog(): Promise<void> {
-    return this.findAndClickButton("Settings");
-  }
-
-  async closeSettingsDialog(): Promise<void> {
-    return this.findAndClickButton("Close");
+  async switchLanguage(locale: string): Promise<void> {
+    // All tests ids so they can be language invariant.
```
Comment (microbit-matt-hillsdon) [inline src/e2e/app.ts:276]:
```suggestion
    // All test ids so they can be language invariant.
```

File: src/e2e/settings.test.ts [hunk 2]
```diff
@@ -11,8 +11,11 @@ describe("Browser - settings", () => {
   afterEach(app.screenshot.bind(app));
   afterAll(app.dispose.bind(app));
 
-  it("shows the settings dialog when the button is clicked", async () => {
```
Comment (microbit-matt-hillsdon) [inline src/e2e/settings.test.ts:14]: This pre-existing test was broken but we didn't notice as the file name didn't match the pattern the test runner uses. It didn't do anything worthwhile so I've just removed it. It dated to when settings was a top-level menu.

**Review by microbit-robert — APPROVED** [review]
LGTM.

**Comment by microbit-matt-hillsdon** [comment]
Spotted an issue where the project name doesn't seem to make it to session storage, so doesn't survive a refresh or tab close/reopen. It's not new though; it affects main too.  Raised #547.

**Pull request merged**